src/auth_identity.rs

```diff
@@ -170,6 +170,14 @@ impl AuthIdentityBuffers {
     pub fn is_empty(&self) -> bool {
         self.user.is_empty()
     }
+
+    pub fn from_utf8(user: &str, domain: &str, password: &str) -> Self {
+        Self {
+            user: utils::string_to_utf16(user),
+            domain: utils::string_to_utf16(domain),
+            password: utils::string_to_utf16(password).into(),
+        }
+    }
```

Comment:
thought: I think we should cleanup our UTF-16-related code using the widestring crate. It would be easier and less error-prone (better defined) than working with `Vec<u8>`.

Reply:
Makes sense